TcgPocket/Features/Cards/commands/UpdateCardCommand.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AutoMapper;
+using FluentValidation;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using TcgPocket.Data;
+using TcgPocket.Features.CardTypes;
+using TcgPocket.Features.Games;
+using TcgPocket.Features.Rarities;
+using TcgPocket.Features.Sets;
+using TcgPocket.Shared;
+using TcgPocket.Shared.Interfaces;
+
+namespace TcgPocket.Features.Cards.Commands
+{
+    public class UpdateCardCommand : IRequest<Response<CardGetDto>>
+    {
+        public int Id { get; set; }
+        public CardDto CardDto { get; set; }
+    }
+
+    public class UpdateCardCommandHandler : IRequestHandler<UpdateCardCommand, Response<CardGetDto>>
+    {
+        private readonly DataContext _dataContext;
+        private readonly IMapper _mapper;
+        private readonly IValidator<UpdateCardCommand> _validator;
+
+        public UpdateCardCommandHandler(DataContext dataContext, IMapper mapper, IValidator<UpdateCardCommand> validator)
+        {
+            _dataContext = dataContext;
+            _mapper = mapper;
+            _validator = validator;
+        }
+
+        public async Task<Response<CardGetDto>> Handle(UpdateCardCommand command, CancellationToken cancellationToken)
+        {
+            var result = await _validator.ValidateAsync(command);
+
+            if (!result.IsValid)
+            {
+                var errors = _mapper.Map<List<Error>>(result.Errors);
+                return new Response<CardGetDto>{ Errors = errors };
+            }
+
+            if (!await _dataContext.Set<Set>().AnyAsync(x => x.Id == command.CardDto.SetId))
+            {
+                return Error.AsResponse<CardGetDto>("Set not found", "setId");
+            }
+            if (!await _dataContext.Set<Rarity>().AnyAsync(x => x.Id == command.CardDto.RarityId))
+            {
+                return Error.AsResponse<CardGetDto>("Rarity not found", "rarityId");
+            }
+            if (!await _dataContext.Set<Game>().AnyAsync(x => x.Id == command.CardDto.GameId))
+            {
+                return Error.AsResponse<CardGetDto>("Game not found", "gameId");
+            }
+            if (!await _dataContext.Set<CardType>().AnyAsync(x => x.Id == command.CardDto.CardTypeId))
+            {
+                return Error.AsResponse<CardGetDto>("Card Type not found", "cardTypeId");
+            }
```

Comment:
It would be nicer on the user here if we returned all of the errors at once instead of one at a time
```suggestion
            var errors = new List<Errors>();

            if (!await _dataContext.Set<Set>().AnyAsync(x => x.Id == command.CardDto.SetId))
            {
                errors.Add(new Error("Set not found", "setId"));
            }
            if (!await _dataContext.Set<Rarity>().AnyAsync(x => x.Id == command.CardDto.RarityId))
            {
                errors.Add(new Error("Rarity not found", "rarityId"));
            }
            if (!await _dataContext.Set<Game>().AnyAsync(x => x.Id == command.CardDto.GameId))
            {
                errors.Add(new Error("Game not found", "gameId"));
            }
            if (!await _dataContext.Set<CardType>().AnyAsync(x => x.Id == command.CardDto.CardTypeId))
            {
                errors.Add(new Error("Card Type not found", "cardTypeId"));
            }
```

Reply:
100%, i forgor we could add all the errors to the response and return them